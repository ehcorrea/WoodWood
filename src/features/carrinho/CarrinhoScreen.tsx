import { View } from 'react-native';

export default function CarrinhoScreen() {
  return <View></View>;
}
