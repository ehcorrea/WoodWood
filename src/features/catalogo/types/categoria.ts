import { Produto } from '@/services/api';

export type ProdutosEmCategoria = Record<string, Produto[]>;
