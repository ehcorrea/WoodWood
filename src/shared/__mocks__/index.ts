export * from './stores';
export * from './produtos';
