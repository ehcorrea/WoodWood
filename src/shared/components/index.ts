export * from './Provider/Provider';
export * from './Text/Text';
