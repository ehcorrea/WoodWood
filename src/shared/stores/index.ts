export * from './carrinhoStore';
export * from './create';
