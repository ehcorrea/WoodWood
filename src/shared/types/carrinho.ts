import { Produto } from '@/services/api';

export type ProdutoSimplificao = Omit<
  Produto,
  'rating' | 'description' | 'category'
>;

export type ProdutoDoCarrinho = ProdutoSimplificao & {
  quantidade: number;
  total: number;
};
