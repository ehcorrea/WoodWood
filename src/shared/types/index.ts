export * from './carrinho';
export * from './theme';
