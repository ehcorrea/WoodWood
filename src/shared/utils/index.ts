export * from './carrinho/carrinho';
export * from './responsive/responsive';
