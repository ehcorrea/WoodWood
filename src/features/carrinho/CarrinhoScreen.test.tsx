import { View as MockedView } from 'react-native';

import { ProdutoDoCarrinho } from '@/shared/types';
import {
  criarMockProdutoDoCarrinho,
  mockedCarrinhoStore,
} from '@/shared/__mocks__';

import { render } from '@/test/utils';

import CarrinhoScreen from './CarrinhoScreen';

jest.mock('./components', () => ({
  CarrinhoVazio: () => <MockedView testID="carrinhoVazio" />,
  CardProduto: () => <MockedView testID="cardProduto" />,
}));

const mockedProduto = criarMockProdutoDoCarrinho({
  quantidade: 3,
  total: 300,
  price: 100,
});

const setup = (produtos: ProdutoDoCarrinho | undefined, total = 300) => {
  mockedCarrinhoStore({
    produtos: {
      [mockedProduto.id]: produtos,
      23: undefined,
    },
    total,
  });
  return render(<CarrinhoScreen />);
};

describe('<CarrinhoScreen/>', () => {
  describe('quando renderizado', () => {
    test('e possui algum produto no carrinho', () => {
      const container = setup(mockedProduto);
      expect(container.getAllByTestId('cardProduto').length).toBe(1);
      expect(container.getByText('Total: R$ 300.00')).toBeTruthy();
      expect(container.queryByTestId('carrinhoVazio')).toBeFalsy();
      expect(container).toMatchSnapshot();
    });
    test('e nÃ£o possui nenhum produto no carrinho', () => {
      const container = setup(undefined, 0);
      expect(container.queryAllByTestId('cardProduto').length).toBe(0);
      expect(container.queryByText('Total: R$ 0.00')).toBeNull();
      expect(container.getByTestId('carrinhoVazio')).toBeTruthy();
      expect(container).toMatchSnapshot();
    });
  });
});
