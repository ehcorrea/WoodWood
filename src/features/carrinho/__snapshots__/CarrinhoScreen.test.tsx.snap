// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarrinhoScreen/> quando renderizado e n√£o possui nenhum produto no carrinho 1`] = `
<View
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": "5%",
        "paddingLeft": "5%",
        "paddingRight": "5%",
        "paddingTop": "5%",
      },
      undefined,
    ]
  }
>
  <View
    testID="carrinhoVazio"
  />
</View>
`;

exports[`<CarrinhoScreen/> quando renderizado e possui algum produto no carrinho 1`] = `
<View
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": "5%",
        "paddingLeft": "5%",
        "paddingRight": "5%",
        "paddingTop": "5%",
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flexBasis": 0,
          "flexGrow": 4,
          "flexShrink": 1,
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      data={
        [
          {
            "id": 0,
            "image": "mocked-image",
            "price": 100,
            "quantidade": 3,
            "title": "mocked-title",
            "total": 300,
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      style={
        {
          "flex": 1,
        }
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View
            testID="cardProduto"
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#1A1D26",
            "fontFamily": "SofiaPro-Regular",
            "fontSize": 21.5,
            "lineHeight": 25.5,
          },
          undefined,
        ]
      }
    >
      Total: R$ 
      300.00
    </Text>
  </View>
</View>
`;
